### PR TITLE
feat: add copy component

### LINKED ISSUE #136: Show visual feedback after copying a code block
After clicking on "Copy", we want to show the user visual feedback.

Github does it like this:

https://user-images.githubusercontent.com/7814926/133559909-c80b1b41-d688-433f-abfb-22fea2ed94b6.mp4

@georgiee already did some experiments, which you can read here: https://github.com/satellytes/satellytes.com/pull/130#issuecomment-913462511

### PR DESCRIPTION
**Pull request by milljoniaer**
### What is included?
This adds a Copy component which shows visual feedback for a few seconds once clicked and copies the passed text.

I already used it in the legacy Code Block Component, so it is displayed in the Blogposts.

https://user-images.githubusercontent.com/78978542/159286255-794cabcd-cf8c-4980-b2c3-c72612e36bba.mov

closes #136

### CODE REVIEW

File: src/components/ui/copy/copy.tsx
Comment (feedm3) [inline src/components/ui/copy/copy.tsx]: Constants always upper-case: `DEFAULT_CLICKED_DURATION`.

Also not 100% happy with the name. It doesn't clearly indicate what it does and the unit is missing. 
Better could be: `SHOW_SUCCESS_NOTIFICATION_MS`

File: src/components/ui/copy/copy.tsx
Comment (feedm3) [inline src/components/ui/copy/copy.tsx]: This is super hard to maintain. Once you change the variable, the comment doesn't get updated automatically. It's probably better to just remove it. As the variable is also on top of the file it's already easy to spot the value.

File: src/components/ui/copy/copy.tsx
Comment (feedm3) [inline src/components/ui/copy/copy.tsx]: Maybe `Copy` is a bit too general, it can mean multiple things (could be text that gets displayed or a more technical copy of a component, etc.). I think `CopyButton` narrows it down to what it actually is.

File: src/assets/locales/de/translations.json
Comment (georgiee) [inline src/assets/locales/de/translations.json:199]: Thanks for thinking of the translations 🙌

**Review by feedm3 — APPROVED** [review]
Good job! I liked that you put the component now into a separate file, very good to read :) 

I have some smaller recommendations. Will already approve so you can merge once you added them.

File: src/components/legacy/icons/markdown-icons/copy.tsx
Comment (georgiee) [inline src/components/legacy/icons/markdown-icons/copy.tsx:1]: thanks for removing this icon file 🙌

File: src/components/ui/copy/copy.stories.mdx
Comment (georgiee) [inline src/components/ui/copy/copy.stories.mdx]: Thanks for the story ✨

https://deploy-preview-504--satellytes-website-storybook.netlify.app/?path=/docs/components-ui-copy--code&globals=locale:en;locales.en:English;locales.de:Deutsch

File: src/components/ui/copy/copy.tsx
Comment (georgiee) [inline src/components/ui/copy/copy.tsx]: I think you don't need the `id` here? It's not referenced at all?

File: src/components/ui/copy/copy.tsx
Comment (georgiee) [inline src/components/ui/copy/copy.tsx]: constants should always be in all caps: `DEFAULT_CLICK_DURATION`

File: src/components/ui/copy/copy.tsx
Comment (georgiee) [inline src/components/ui/copy/copy.tsx]: This name doesn't resonate with me, what about `feedbackMessageTime`? The `click` is of course the action someone did before but describing the thing, the "feedback message" makes it more readable. You can also omit `copy` because that's the context of the entire component anyway.

File: src/components/legacy/markdown/code-block.tsx
Comment (georgiee) [inline src/components/legacy/markdown/code-block.tsx]: Even though code-block is legacy code it's okay to add it. Let's tackle and eliminate the legacy folder soon afterwards instead.

File: src/components/ui/copy/copy.tsx
Comment (georgiee) [inline src/components/ui/copy/copy.tsx]: so that's the magic that shows the clicked message. While we don't have actual side effects here I wonder if involving `useEffect` would be the more correct approach. Maybe someone from the crew has some opinion on that. 

I'm happy with the given solution 👍

**Review by georgiee — CHANGES_REQUESTED** [review]
Nice, this looks great. I have provided a few comments and a few things to change (constant related, variable names) but nothing with big issues.

I will lock this PR until handled and then we can merge. Thank you 🙌

**Comment by milljoniaer** [comment]
@georgiee @feedm3 Thank you for your recommendations, I will implement them now.

But I have a remaining question:
With the legacy component we copied the text with a "copy-to-clipboard" dependency.
I have now used navigator.clipboard.writeText(). Is that fine? And should I remove this dependency?

**Comment by feedm3** [comment]
The native function is supported by all major browsers. I'm not sure why the dependency was used, maybe you can check if it has some important features that we want to have. Your implementation seems fine 

https://caniuse.com/mdn-api_clipboard_writetext

**Comment by georgiee** [comment]
> @georgiee @feedm3 Thank you for your recommendations, I will implement them now.
> 
> But I have a remaining question: With the legacy component we copied the text with a "copy-to-clipboard" dependency. I have now used navigator.clipboard.writeText(). Is that fine? And should I remove this dependency?

As said and linked by @feedm3 I think removing the dependency is the way to go. I checked the git history and found this commit https://github.com/satellytes/satellytes.com/commit/8983e403f49334e4dafd99020cc973769ec2a7b2 where it was introduced. I guess there is no particular reason, maybe we did not think of the native functionality back then.

File: src/components/ui/copy/copy.tsx
Comment (milljoniaer) [reply src/components/ui/copy/copy.tsx]: Ohs 🧹

File: src/components/ui/copy/copy.tsx
Comment (milljoniaer) [reply src/components/ui/copy/copy.tsx]: 🧹

File: src/components/ui/copy/copy.tsx
Comment (milljoniaer) [reply src/components/ui/copy/copy.tsx]: 👍

File: src/components/ui/copy/copy.tsx
Comment (milljoniaer) [reply src/components/ui/copy/copy.tsx]: Ohps, leftover of one of my previous experiments

File: src/components/ui/copy/copy.tsx
Comment (milljoniaer) [reply src/components/ui/copy/copy.tsx]: @feedm3 suggested showSuccessNotificationMS, I have now used that

File: src/components/legacy/markdown/code-block.tsx
Comment (milljoniaer) [reply src/components/legacy/markdown/code-block.tsx]: 👍

File: src/components/ui/copy/copy.tsx
Comment (milljoniaer) [reply src/components/ui/copy/copy.tsx]: so do we leave it as it is, or has anyone thoughts on that?

**Comment by milljoniaer** [comment]
Thank you for your comments, I pushed a new version and left some comments, please have another look

File: src/components/ui/copy/copy.tsx
Comment (georgiee) [reply src/components/ui/copy/copy.tsx]: @feedm3 maybe? otherwise leave it as is.

File: src/components/ui/copy/copy.tsx
Comment (georgiee) [reply src/components/ui/copy/copy.tsx]: thanks! Mind the `MS` postfix noting the unit format of milliseconds. That's a good point too 👍 I agree with the name.

File: package.json
Comment (georgiee) [inline package.json:42]: byeee 👋

**Review by georgiee — APPROVED** [review]
Thanks for your changes. Let's merge it 🙏

**Pull request merged**
